### PR TITLE
[mono-api-html] Print ref/out modifiers on parameters. Fixes #59430.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Print ref/out modifiers on parameters, so that any changes in ref/out are detected.

https://bugzilla.xamarin.com/show_bug.cgi?id=59430

### CODE REVIEW

File: mcs/tools/mono-api-html/ConstructorComparer.cs [hunk 1]
```diff
@@ -133,10 +133,15 @@ public override string GetDescription (XElement e)
...
-					var pTypeName   = p.GetTypeName ("type");
-					list.Add (State.IgnoreParameterNameChanges
-						? pTypeName
-						: pTypeName + " " + p.GetAttribute ("name"));
+					var param = p.GetTypeName ("type");
+					if (!State.IgnoreParameterNameChanges)
+						param += " " + p.GetAttribute ("name");
+
+					var direction = p.GetAttribute ("direction");
+					if (direction?.Length > 0)
```
Comment (akoeplinger) [inline mcs/tools/mono-api-html/ConstructorComparer.cs]: I'd use `!String.IsNullOrEmpty (direction)` here but that's just personal preference

**Comment by marek-safar** [comment]
This is going to break your past apis comparison. @jonpryor @spouliot are you ok with it?

**Comment by rolfbjarne** [comment]
@marek-safar how would this break comparison? Two members that differ by ref/out/none should not be considered identical in the first place (which they currently are).

So yes, this may cause additional changes to show up, but that's the purpose of the patch (find those changes).

**Comment by marek-safar** [comment]
@rolfbjarne it's used for comparison hence any ref or out member will show up as a new member

**Comment by rolfbjarne** [comment]
Correct, comparing two methods that differ by ref/out/none will show up as method removed + method added.

Arguably it could be better rendered (a single method and inline showing that the ref/out/none changed), but that looked like a bigger change, and this way would still fix the bug (ref/out changes didn't show up at all).

**Comment by marek-safar** [comment]
I am ok with it but still want a confirmation from @jonpryor and @spouliot

**Pull request merged**